### PR TITLE
feat(ai-server): integrate google sheets and update deployment

### PR DESCRIPTION
**Pull request by y3rsh**
# Overview
<https://opentrons.atlassian.net/browse/AUTH-936>
Add a Google sheets integration so store the input of calls to the feedback API endpoint.

When this is merged to edge the CD is expected to create the new task/container definition correctly as the environment variables and secrets have been added.

## Test Plan and Hands on Testing

- [x] direct API calls using live client successfully store data
- [x] feedback through the UI successfully stores data
- [x] both types of tokens are handled with the new User object
- [x] Add staging and prod secrets and environment variables in AWS
- [x] update deploy script to reconcile secrets in the container definition
- [x] document

## Local validation
`AWS_PROFILE=robotics_ai_staging make dry-deploy ENV=staging`
and
`AWS_PROFILE=robotics_ai_prod make dry-deploy ENV=prod`
Both output the below as expected:
```log
Expected secrets: {'HUGGINGFACE_API_KEY', 'OPENAI_API_KEY', 'DATADOG_API_KEY', 'GOOGLE_CREDENTIALS_JSON'}
Existing secrets: {'HUGGINGFACE_API_KEY', 'OPENAI_API_KEY', 'DATADOG_API_KEY'}
Missing secrets: ['GOOGLE_CREDENTIALS_JSON']
Adding missing secret: GOOGLE_CREDENTIALS_JSON
```

## TLDR AI Summary

This pull request introduces a new feature for Google Sheets integration, updates the authentication mechanism, and refines several API endpoints to enhance functionality and security. The most important changes include the addition of a Google Sheets client, modifications to the authentication process to return a `User` object, and updates to various API endpoints to include user information and handle feedback.

### Google Sheets Integration:
* [`opentrons-ai-server/Makefile`](diffhunk://#diff-29956e3652f7931a8fa89e5f1f3702d497a81cf98ed2115b200208f99244f326R136-R140): Added a `test-googlesheet` target to run Google Sheets integration tests.
* [`opentrons-ai-server/Pipfile`](diffhunk://#diff-fc8b848af46eed2e3c916ec75cdff28b9ede04651a8813eb3e70ba11fa1de7a2R20-R22): Added dependencies for `gspread`, `google-auth`, and `google-auth-oauthlib`.
* [`opentrons-ai-server/api/integration/google_sheets.py`](diffhunk://#diff-c273cda81a6fac186f769fd1b3763589b2e67ed874fcf05bab8adb8d4527c562R1-R74): Added a new `GoogleSheetsClient` class to handle Google Sheets operations.
* [`opentrons-ai-server/README.md`](diffhunk://#diff-f036d779527f141103ef11175f9de251cdc212a196b63ed35e156a29e4dbd89aR120-R137): Updated documentation to include instructions for setting up Google Sheets integration.

### Authentication Enhancements:
* [`opentrons-ai-server/api/integration/auth.py`](diffhunk://#diff-2711b4f4e6f16f8b030b72dda2873999d92b6a069baf0d4edfabf4bb53123bb6L1-R6): Modified the `verify` method to return a `User` object instead of a generic payload. [[1]](diffhunk://#diff-2711b4f4e6f16f8b030b72dda2873999d92b6a069baf0d4edfabf4bb53123bb6L1-R6) [[2]](diffhunk://#diff-2711b4f4e6f16f8b030b72dda2873999d92b6a069baf0d4edfabf4bb53123bb6L31-R31) [[3]](diffhunk://#diff-2711b4f4e6f16f8b030b72dda2873999d92b6a069baf0d4edfabf4bb53123bb6L53-R54)
* [`opentrons-ai-server/api/models/user.py`](diffhunk://#diff-2053a57faafa7c673687ed89f088d086453ff05c58d5457f8d74c12dca08c815R1-R16): Introduced a new `User` model to represent authenticated users.

### API Endpoint Updates:
* [`opentrons-ai-server/api/handler/fast.py`](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dL4-R11): Updated various endpoints (`create_chat_completion`, `update_protocol`, `create_protocol`, `feedback`) to include user information and handle feedback using the `GoogleSheetsClient`. [[1]](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dL4-R11) [[2]](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dR24-R33) [[3]](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dR44) [[4]](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dL178-R190) [[5]](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dL201-R207) [[6]](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dL220-R247) [[7]](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dL260-R272) [[8]](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dL282-R288) [[9]](diffhunk://#diff-1396285c1c56820f9e2c284f104ff5d575be085455d489df8b7dea7e54d2019dL342-R358)
* [`opentrons-ai-server/api/models/feedback_request.py`](diffhunk://#diff-aefa781505c05c0efe5f5f64e7641265f54901af6364fc8c764bc20e9ce21f0dR1-R15): Added a new `FeedbackRequest` model to validate feedback input.

### Configuration Updates:
* [`opentrons-ai-server/api/settings.py`](diffhunk://#diff-760056bbff3145b44e027b5c1d5dbe0b9efd275278b150d11cb1b04f15928f8aR37-R44): Added new settings for Google Sheets integration, including `google_sheet_id`, `google_sheet_worksheet`, and `google_credentials_json`.

### Testing Enhancements:
* [`opentrons-ai-server/tests/helpers/client.py`](diffhunk://#diff-d08f1dc39e5d41c66ea108eb8ad13a2eb1786db8d00620a2838ea9e4d37932acR6): Updated the test client to include a method for posting feedback. [[1]](diffhunk://#diff-d08f1dc39e5d41c66ea108eb8ad13a2eb1786db8d00620a2838ea9e4d37932acR6) [[2]](diffhunk://#diff-d08f1dc39e5d41c66ea108eb8ad13a2eb1786db8d00620a2838ea9e4d37932acL71-R78)

### CODE REVIEW

File: opentrons-ai-server/README.md [hunk 1]
```diff
@@ -117,3 +126,48 @@ The live-test target will run tests against any environment. The default is loca
...
+
+1. Create a Google Cloud Platform project
+1. Enable the Google Sheets and Drive API
+1. Go to APIs & Services > Library and enable the Google Sheets API.
+1. Go to APIs & Services > Credentials and create a Service Account. This account will be used by your application to access the Google Sheets API.
+1. After creating the Service Account, click on it in the Credentials section, go to the Keys tab, and create a JSON key. This will download a JSON file with credentials for your Service Account.
+1. Open the JSON file and store its content securely. You’ll set this JSON content as an environment variable.
+1. Configure Access to the Google Sheet
+1. Open the Google Sheet you want to access.
+1. Click Share and add the Service Account email (found in the JSON file under "client_email") as a collaborator, typically with Editor access. This allows the Service Account to interact with the sheet.
```
Comment (Elyorcv) [inline opentrons-ai-server/README.md:140]: Is this something every user needs to do?

File: opentrons-ai-server/api/handler/fast.py [hunk 2]
```diff
@@ -257,15 +258,15 @@ async def update_protocol(
     description="Generate a chat response based on the provided prompt that will create a new protocol with the required changes.",
 )
 async def create_protocol(
-    body: CreateProtocol, auth_result: Any = Security(auth.verify)  # noqa: B008
+    body: CreateProtocol, user: Annotated[User, Security(auth.verify)]
 ) -> Union[ChatResponse, ErrorResponse]:  # noqa: B008
     """
-    Generate an updated protocolusing OpenAI.
+    Generate an updated protocol using OpenAI.
```
Comment (Elyorcv) [inline opentrons-ai-server/api/handler/fast.py:264]: maybe just LLM or model, since we soon change to different models?

**Review by Elyorcv — APPROVED** [review]
LGTM
